app/models/post.rb

```diff
@@ -0,0 +1,38 @@
+class Post < ActiveRecord::Base
+  belongs_to :user
+  has_many :comments, dependent: :destroy
+
+  has_many :favourites, dependent: :destroy
+  has_many :favourited_users, through: :favourites, source: :user
+
+  has_many :post_tag_links, dependent: :destroy
+  has_many :tags, through: :post_tag_links
+
+  validates :title, presence: true, uniqueness: true
+  paginates_per 10
+
+  def user_name
+    if user
+      user.user_name
+    else
+      "Anonymous"
+    end
+  end
+
```

Comment:
You have user_name defined in a couple spots and one time it is defined as above, but in the user model you have it defined as "#{first_name} #{last_name}". I'm guessing it didn't cause you any problems but it might be a decent idea not to reuse the method name in both cases?